h2o-py/h2o/plot/_matplotlib.py

```diff
@@ -5,6 +5,8 @@ def get_matplotlib_pyplot(server, raise_if_not_available=False):
         # noinspection PyUnresolvedReferences
         import matplotlib
         if server:
+            import matplotlib.pyplot as plt
+            plt.close('all')
```

Comment:
I have some concerns about this. It might be ok but have you tested the explanations with `render=False`? (https://github.com/h2oai/h2o-3/blob/master/h2o-py/h2o/explanation/_explain.py#L3089)

The goal there is to have an object that can be rendered later and if every call to `get_matplotlib_pyplot` closes the figures I think this won't work.

In one jupyter cell do
```python
ex = aml.explain(test_frame, render=False)
```

and in another
```python
ex
```

or 

```python
display(ex)
```

It should render the same thing as if you'd call `aml.explain(test_frame)`.

Reply:
Maybe a workaround would be to check if the backend is `Agg` and if it is do nothing, otherwise do what you did here. Explain calls multiple plot functions and every single plot function call this function but once it's set to `Agg` it should not change for any subsequent plots.

Reply:
I tested the scenario that I thought could be problematic and it works. I didn't do extensive testing over multiple python/matplotlib versions but I think it should be ok. Thank you for notifying me, @krasinski !